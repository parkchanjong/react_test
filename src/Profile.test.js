import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile username="chanjong" name="박찬종" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders username and name", () => {
    const wrapper = mount(<Profile username="chanjong" name="박찬종" />);

    expect(wrapper.props().username).toBe("chanjong");
    expect(wrapper.props().name).toBe("박찬종");

    const boldElement = wrapper.find("b");
    expect(boldElement.contains("chanjong")).toBe(true);
    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("(박찬종)");
  });
});
