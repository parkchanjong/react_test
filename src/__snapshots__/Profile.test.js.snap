// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="박찬종"
  username="chanjong"
>
  <div>
    <b>
      chanjong
    </b>
     
    <span>
      (
      박찬종
      )
    </span>
  </div>
</Profile>
`;
